Hazel/Dtls/Record.cs

```diff
@@ -53,7 +58,7 @@ record = new Record();
             record.SequenceNumber = span.ReadBigEndian48(5);
             record.Length = span.ReadBigEndian16(11);
 
-            if (version != ProtocolVersion.DTLS1_2)
+            if (expectedProtocolVersion.HasValue && version != expectedProtocolVersion.Value)
```

Comment:
Same kind of mention, we should start storing the record version in this method.

Reply:
For this one, let me double check with you - I added the record version on the record struct, but the `expectedProtocolVersion` comes from connection meta data stored on `PeerData`, so I think the expected version should be passed in as it is here? Let me know to confirm - thank you!